libs/libarchfpga/src/scatter_gather_types.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <string>
+#include "switchblock_types.h"
+
+/**
+ * @brief Enumeration for the type field of an <sg_pattern> tag.
```

Comment:
Explain what it controls ... as in the reference, explain that with unidir the gather pattern makes a mux which is connected by a node to the scatter pattern of edges. With bidir, the structure is made symmetric with a gather mux and a scatter edge pattern on each end of the node linking them.